cmd/snap-update-ns/utils.go

```diff
@@ -87,6 +89,49 @@ func IsReadOnly(dirFd int, dirName string) (bool, error) {
 	return false, nil
 }
 
+// IsTrustedTmpfs returns true if a directory belongs to a tmpfs mounted by snapd.
+//
+// The function inspects the directory (represented as both an open file
+// descriptor and the absolute path) and a list of changes that were applied to
+// the mount namespace. A directory is trusted if it is a tmpfs that was
+// mounted by snap-confine or snapd-update-ns. Note that sub-directories of a
+// trusted tmpfs are not considered trusted by this function.
+func IsTrustedTmpfs(dirFd int, dirName string, changes []*Change) (bool, error) {
+	var fsData syscall.Statfs_t
+	if err := sysFstatfs(dirFd, &fsData); err != nil {
+		return false, fmt.Errorf("cannot fstatfs %q: %s", dirName, err)
+	}
+	// If something is not a tmpfs it cannot be the trusted tmpfs we are looking for.
+	if fsData.Type != TmpfsMagic {
+		return false, nil
+	}
+	// Any of the past changes that mounted a tmpfs exactly at the directory we
+	// are inspecting is considered as trusted. This is conservative because it
+	// doesn't trust sub-directories of a trusted tmpfs. This approach is
+	// sufficient for the intended use.
+	//
+	// The algorithm goes over all the changes and picks up mount and unmount
+	// actions keeping track of what happened. The set of constraints in
+	// snap-update-ns and snapd prevent from mounting over an existing mount
+	// point so we don't need to consider e.g. a bind mount shadowing an active
+	// tmpfs.
+	mounted := false
+	for _, change := range changes {
+		if change.Entry.Type == "tmpfs" && change.Entry.Dir == dirName {
+			mounted = change.Action == Mount
+		}
+	}
```

Comment:
This is fine as is so not a blocker. It looks like you are simply taking the last tmpfs whose name matches dirName and taking the last action. Would it be better to go backwards through the changes, do the same check/assignment but instead 'break' to achieve a small optimization?